packages/core/src/utils/schema.ts

```diff
@@ -22,11 +23,24 @@ export const BigNumberishRangeSchema = z
   })
   .strict();
 
-export const PercentageSchema = z
+export const MultiplierSchema = z
   .object({
     /**
-     * Percent value between 1 and 1000 inclusive
+     * Multiplier value with max precision of 4 decmial places
      */
-    percentage: z.number().min(1).max(1000),
+    multiplier: z.number().refine(
+      (n) => {
+        return n.toString().split(".")[1].length <= 4;
```

Comment:
would actually love thoughts on this. Wanted to cap this here so we don't accidentally get a super long decimal value that we then overflow in the `bigIntMultiply` function.

Reply:
this makes sense to me

Reply:
same. makes sense to me.